app/src/main/java/com/google/android/ground/ui/tos/TermsOfServiceFragment.kt

```diff
@@ -41,13 +66,62 @@ class TermsOfServiceFragment : AbstractFragment() {
     inflater: LayoutInflater,
     container: ViewGroup?,
     savedInstanceState: Bundle?,
-  ): View {
-    val args = TermsOfServiceFragmentArgs.fromBundle(requireArguments())
-    val binding = FragmentTermsServiceBinding.inflate(inflater, container, false)
-    binding.viewModel = viewModel
-    binding.isViewOnly = args.isViewOnly
-    binding.lifecycleOwner = this
-    return binding.root
+  ): View = createComposeView {
+    CreateView(TermsOfServiceFragmentArgs.fromBundle(requireArguments()))
+  }
+
+  @Composable
+  private fun CreateView(args: TermsOfServiceFragmentArgs) {
+    Scaffold(
+      topBar = {
+        Toolbar(
+          stringRes = R.string.tos_title,
+          isBackArrowVisible = args.isViewOnly,
+          iconClick = { findNavController().navigateUp() },
+        )
+      }
+    ) { innerPadding ->
+      val termsText by viewModel.termsOfServiceText.observeAsState(AnnotatedString(""))
+      val agreeChecked by viewModel.agreeCheckboxChecked.observeAsState(false)
+
+      Column(
+        modifier =
+          Modifier.padding(innerPadding)
+            .fillMaxSize()
+            .padding(16.dp)
+            .verticalScroll(rememberScrollState()),
+        horizontalAlignment = Alignment.CenterHorizontally,
+      ) {
+        Spacer(modifier = Modifier.height(16.dp))
+        Text(text = termsText.toString(), modifier = Modifier.padding(8.dp))
+        Spacer(modifier = Modifier.height(16.dp))
+
+        if (!args.isViewOnly) {
+          Row(verticalAlignment = Alignment.CenterVertically, modifier = Modifier.fillMaxWidth()) {
+            Checkbox(
+              modifier = Modifier.testTag("agreeCheckBox"),
+              checked = agreeChecked,
+              onCheckedChange = { viewModel.agreeCheckboxChecked.value = it },
+            )
+            Text(
+              text = stringResource(R.string.agree_checkbox),
+              modifier = Modifier.clickable { viewModel.agreeCheckboxChecked.value = !agreeChecked },
+            )
+          }
+
+          Spacer(modifier = Modifier.height(16.dp))
+
+          Button(
+            modifier = Modifier.testTag("agreeButton"),
```

Comment:
Can we use button's text instead of test tag for finding the compose node in tests?